### ISSUE TITLE
Imagick PDF decorators missing on Alpine 3.19

### ISSUE THREAD

**Issue by eddy84** [mlocati/docker-php-extension-installer#857]
### Version of install-php-extensions

latest

### Error description

As written in this issue:
https://github.com/ImageMagick/ImageMagick/issues/6935

Imagick packages were split up into separate packages on alpine:3.19

We can't use imagick with PDFs anymore because imagemagick-pdf  needs to be installed

I had to restrict our docker image to the previous version for now:
php:8.2-fpm-alpine3.18

### Docker image

php:8.2-fpm-alpine

### Minimal Dockerfile

```Dockerfile
ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
RUN chmod uga+x /usr/local/bin/install-php-extensions && sync

# Install php extensions
RUN apk add --no-cache --virtual .build-deps $PHPIZE_DEPS \
#    && apk add --no-cache libedit-dev postgresql-dev libcurl \
    && install-php-extensions \
      bcmath \
      curl \
      exif \
      gd \
      gettext \
      iconv \
      igbinary\
      imagick \
      imap \
      intl \
      mbstring \
      mcrypt \
      opcache \
      pcntl \
      pgsql \
      pdo \
      pdo_pgsql\
      redis \
      xml \
      xsl \
      zip \
    && rm -f /var/cache/apk/*
```

**Comment by hexus** [comment]
Indeed, it would be nice to know why we can't install `imagick` for PHP 8.3. I assume these extra dependencies for Alpine 3.19+ will need to be listed somewhere.

I too will need to stick with PHP 8.2 with Alpine 3.18 as a result, for now.

**Comment by hexus** [comment]
Thanks @mlocati, I landed there in the end after a little searching. Guess we're just waiting for the upstream. 🙂

**Comment by mlocati** [comment]
Indeed.

**Comment by mvorisek** [comment]
Before Alpine 3.18 we had to use this hack https://github.com/mvorisek/image-php/blob/a38b8a1ac588e68e5ed74a59dbd97b17e0b3a710/data/8.2-alpine/Dockerfile#L38 to reduce the image size.

related with upstream change https://gitlab.alpinelinux.org/alpine/aports/-/commit/61dce5f6a84d9f96ab7aa6437f12d2845b572ed8

I belive some basic codes like pdf should be always present on all OS (Debian, Alpine). Not sure is any codec still require the useless Ghostscript CLI app.

**Comment by mateusjatenee** [comment]
Hi — did anyone figure out a workaround for this?

**Comment by lamskoy** [comment]
Would be nice to have PHP code for testing PDFs to make quick test

**Comment by mlocati** [comment]
Isn't PDF already supported?

Here's a sample session in a `php:8.4-cli-alpine` docker image:

```sh
$ install-php-extensions imagick
[...]
$ php -r 'var_export(Imagick::queryFormats());' | grep -i pdf
  48 => 'EPDF',
  155 => 'PDF',
  156 => 'PDFA',
```

**Comment by mlocati** [comment]
Closing because of lack of feedback

**Comment by mvorisek** [comment]
> Hi — did anyone figure out a workaround for this?

Feel free to test https://github.com/mvorisek/image-php - there are many workarounds implemented.

**Comment by mlocati** [comment]
@mvorisek can you explain what's missing in `install-php-extensions`?

I mean, when installing imagick with `install-php-extensions`, we have that `Imagick::queryFormats()` includes `PDF`...

**Comment by mvorisek** [comment]
The Dockerfile is here: https://github.com/mvorisek/image-php/blob/master/data/8.4-alpine/Dockerfile

I do not remember how I fixed the https://github.com/mlocati/docker-php-extension-installer/issues/857#issuecomment-1929187512...

**Comment by mlocati** [comment]
I don't understand what doesn't work... Could you provide a sample PHP script that shows what doesn't work with the way `install-php-extensions` installs `imagick`?
